pkg/handler/collector/collector.go

```diff
@@ -51,16 +52,35 @@ func RegisterDocumentCollector(c Collector, collectorType string) {
 
 // Collect takes all the collectors and starts collecting artifacts
 // after Collect is called, no calls to RegisterDocumentCollector should happen.
-func Collect(ctx context.Context) (<-chan *processor.Document, <-chan error, int, error) {
+func Collect(ctx context.Context, emitter processor.Emitter, handleErr processor.ErrHandler) error {
 	// docChan to collect artifacts
 	docChan := make(chan *processor.Document, BufferChannelSize)
 	// errChan to receive error from collectors
 	errChan := make(chan error, len(documentCollectors))
+
+	var wg sync.WaitGroup
+	wg.Add(len(documentCollectors))
 	for _, collector := range documentCollectors {
 		c := collector
 		go func() {
 			errChan <- c.RetrieveArtifacts(ctx, docChan)
+			wg.Done()
 		}()
 	}
-	return docChan, errChan, len(documentCollectors), nil
+	wg.Wait()
```

Comment:
assuming that this is a semaphore, i dont think we want to wait till everything is done before emitting any documents, or handling the errors.. I think we can copy the original for loop code and channel drain previously in `example.go`?